### PR TITLE
Improve nonexistent decl diagnostic

### PR DESCRIPTION
**Pull request by harlanhaskins**
#### What's in this pull request?

Importing a specific decl from a module fails with `no such decl in module`.
This patch spells out the module that doesn't exist and its type.

#### Resolved bug number: ([SR-](https://bugs.swift.org/browse/SR-))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

A smoke test on macOS does the following:
1. Builds the compiler incrementally.
2. Builds the standard library only for macOS. Simulator standard libraries and
   device standard libraries are not built.
3. lldb is not built.
4. The test and validation-test targets are run only for macOS. The optimized
   version of these tests are not run.

A smoke test on Linux does the following:
1. Builds the compiler incrementally.
2. Builds the standard library incrementally.
3. lldb is built incrementally.
4. The swift test and validation-test targets are run. The optimized version of these
   tests are not run.
5. lldb is tested.

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

File: lib/Sema/NameBinding.cpp [hunk 1]
```diff
@@ -225,9 +225,12 @@ void NameBinder::addImport(
     lookupInModule(topLevelModule, declPath, declPath.front().first, decls,
                    NLKind::QualifiedLookup, ResolutionKind::Overloadable,
                    /*resolver*/nullptr, &SF);
-
+
```
Comment (CodaFi) [inline lib/Sema/NameBinding.cpp]: Nitpick: whitespace.

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
@swift-ci please smoke test Linux platform.

**Comment by jrose-apple** [comment]
I figured this was redundant since it's on the same line as the import. I mean, it's nice when diagnostics show they understand your code, but I'm not convinced this actually communicates that rather than just repeating information.

**Comment by harlanhaskins** [comment]
I think we can hit a sweet spot here. 'No such decl in module' is a really obtuse error.

**Comment by CodaFi** [comment]
@swift-ci please smoke test Linux platform.

**Pull request merged**